crates/sui-e2e-tests/tests/rest.rs

```diff
@@ -55,3 +64,303 @@ async fn execute_transaction_transfer() {
+        vec![sender_sig, sponsor_sig],
+    );
+    let effects = client
+        .execute_transaction(
+            &ExecuteTransactionQueryParameters::default(),
+            &signed_transaction,
+        )
+        .await
+        .unwrap()
+        .effects;
+
+    assert!(effects.status().is_ok());
+    assert_eq!(resolved.effects, effects.into());
+}
+
+#[sim_test]
+async fn resolve_transaction_borrowed_shared_object() {
+    let test_cluster = TestClusterBuilder::new().build().await;
+
+    let client = Client::new(test_cluster.rpc_url());
+
+    let sender = test_cluster.wallet.get_addresses()[0];
+
+    let unresolved_transaction = UnresolvedTransaction {
+        ptb: UnresolvedProgrammableTransaction {
+            inputs: vec![UnresolvedInputArgument::Shared {
+                object_id: "0x6".parse().unwrap(),
+                initial_shared_version: None,
+                mutable: None,
+            }],
+            commands: vec![Command::MoveCall(sui_sdk2::types::MoveCall {
+                package: "0x2".parse().unwrap(),
+                module: "clock".parse().unwrap(),
+                function: "timestamp_ms".parse().unwrap(),
+                type_arguments: vec![],
+                arguments: vec![Argument::Input(0)],
+            })],
+        },
+        sender: sender.into(),
+        gas_payment: None,
+        expiration: TransactionExpiration::None,
+    };
+
+    let resolved = client
+        .inner()
+        .resolve_transaction(&unresolved_transaction)
+        .await
+        .unwrap()
+        .into_inner();
+
+    let signed_transaction = test_cluster
+        .wallet
+        .sign_transaction(&resolved.transaction.into());
+    let effects = client
+        .execute_transaction(
+            &ExecuteTransactionQueryParameters::default(),
+            &signed_transaction,
+        )
+        .await
+        .unwrap()
+        .effects;
+
+    assert!(effects.status().is_ok());
+}
+
+#[sim_test]
+async fn resolve_transaction_mutable_shared_object() {
+    let test_cluster = TestClusterBuilder::new().build().await;
+
+    let client = Client::new(test_cluster.rpc_url());
+
+    let (sender, mut gas) = test_cluster.wallet.get_one_account().await.unwrap();
+    gas.sort_by_key(|object_ref| object_ref.0);
+    let obj_to_stake = gas.first().unwrap().0;
+    let validator_address = client
+        .inner()
+        .get_system_state_summary()
+        .await
+        .unwrap()
+        .inner()
+        .active_validators
+        .first()
+        .unwrap()
+        .address;
+
+    let unresolved_transaction = UnresolvedTransaction {
+        ptb: UnresolvedProgrammableTransaction {
+            inputs: vec![
+                UnresolvedInputArgument::Shared {
+                    object_id: "0x5".parse().unwrap(),
+                    initial_shared_version: None,
+                    mutable: None,
+                },
+                UnresolvedInputArgument::ImmutableOrOwned(UnresolvedObjectReference {
```

Comment:
I was trying to track down our original conversations around modeling these inputs, but something thats missing here is the ability to model inputs where we don't know if they are Immutable/Owned/Shared/Receiving, which will be important to support before this can be used by the sdk

Reply:
What we do in the typescript SDK is to just use a single type that supports providing some of the data: https://sdk.mystenlabs.com/typedoc/interfaces/_mysten_wallet_standard.UnresolvedObjectArg.html